api/examples/src/message/waitForReply.ts

```diff
@@ -0,0 +1,22 @@
+import { GearApi, UserMessageSent } from '@gear-js/api';
+
+export function waitForReply(api: GearApi, programId: string): (messageId: string) => Promise<UserMessageSent> {
+  const messages = {};
+  api.query.system.events((events) => {
+    events.forEach(({ event }) => {
+      if (api.events.gear.UserMessageSent.is(event)) {
+        const {
+          data: {
+            message: { source, reply },
+          },
+        } = event as UserMessageSent;
```

Comment:
It is possible to set the type before without using `as`?

```js
const { some key  }: UserMessageSent = event;
```

Reply:
It's impossible